Realm/RLMResults.mm

```diff
@@ -560,6 +592,15 @@ - (id)objectAtIndex:(NSUInteger)index {
     @throw RLMException(@"Index is out of bounds.", @{@"index": @(index)});
 }
 
+- (id)valueForKey:(NSString *)key {
+    RLMResultsValidate(self);
+    return @[];
+}
+
+- (void)setValue:(__unused id)value forKey:(__unused NSString *)key {
+    RLMResultsValidateInWriteTransaction(self);
```

Comment:
Why not? It would potentially allow the user to think they were doing things on the right thread when they really weren't.